okhttp/src/main/java/okhttp3/internal/http1/Http1Codec.java

```diff
@@ -491,4 +515,19 @@ private void readChunkSize() throws IOException {
       closed = true;
     }
   }
+
+
+  private class Http1ConnectionTimeout extends AsyncTimeout {
+    @Override protected void timedOut() {
+      state = STATE_CANCEL;
```

Comment:
I believe this gets called on a separate thread, does `state` need to be made thread safe?